pkgs/googleapis_firestore_v1/tool/update_protos.dart

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Run this script to update to the latest google/protobuf and googleapis
+/// protobufs.
+library;
+
+import 'dart:io';
+
+void main(List<String> args) async {
+  final dartToolDir = Directory('.dart_tool');
+  final protobufDir = Directory('.dart_tool/protobuf');
+  final googleapisDir = Directory('.dart_tool/googleapis');
+
+  final destDir = Directory('protos');
+
+  if (protobufDir.existsSync()) {
+    await git(['pull'], cwd: protobufDir);
```

Comment:
Should we also use `--depth 1` here?

```suggestion
    await git(['pull', '--depth', '1'], cwd: protobufDir);
```

Reply:
Ah! I was unaware that was an option on `git pull`; sgtm.